### ISSUE TITLE
[BUG] There's a VM stuck in terminating after upgrade

### ISSUE THREAD

**Issue by noahgildersleeve** [harvester/harvester#3529]
**Describe the bug**

After finishing an upgrade from `v1.1.1` to `v1.1.2-rc1` there's a VM that is still showing up as terminating. You won't see it in the Harvester VM list because the namespace is harvester-system, but you can see it if you check via embedded Rancher in the explorer.
**To Reproduce**
1. Create a single node cluster with v1.1.1
2. Upgrade to v1.1.2
3. Finish upgrade
4. Check for VMs in harvester-system namespace
**Expected behavior**

The VM should finish terminating and delete after upgrade
**Support bundle**

[support bundle still showing upgrade thing.zip](https://github.com/harvester/harvester/files/10808402/support.bundle.still.showing.upgrade.thing.zip)

**Environment**
 - Harvester ISO version: v1.1.2-rc1
 - Rancher version: Single node docker v2.7.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): libvirt single node Harvester VM with 

**Additional context**
This was seen on a VM that had been prepped with all of the pre-upgrade tasks. I'm going to try to repro with the minimum setup.
![Screenshot_20230222_130812](https://user-images.githubusercontent.com/83787952/220760949-6607ba31-5419-419c-924b-0d15aa41d9e6.png)

**Comment by lanfon72** [comment]
update labels and change `severity` to `1` as this bug will affect cluster stability when scaling.

could be reproduced on Rancher both `v2.7.1` and `v2.6-5d91d1afdd7d88da6c0530080b16a5ad6dae8d17-head` with Harvester v1.1.2-rc1 (upgraded from `v1.1.0` or `v1.1.1`)

**Comment by bk201** [comment]
This should be caused by https://github.com/harvester/harvester/issues/2677, the VirtualMachine resource exists due to the `wrangler.cattle.io/VMController.UnsetOwnerOfPVCs` finalizer (VMI and virt-launcher pod doesn't exist).
```
kind: VirtualMachine
metadata:
  annotations:
    harvesterhci.io/removedPersistentVolumeClaims: upgrade-repo-hvst-upgrade-p9w6j-disk-0
    harvesterhci.io/vmRunStrategy: Always
    harvesterhci.io/volumeClaimTemplates: '[{"metadata":{"name":"upgrade-repo-hvst-upgrade-p9w6j-disk-0","creationTimestamp":"null","annotations":{"harvesterhci.io/imageId":"harvester-syste
m/harvester-iso-ffrk2"}},"spec":{"accessModes":["ReadWriteMany"],"resources":{"requests":{"storage":"10Gi"}},"storageClassName":"longhorn-harvester-iso-ffrk2","volumeMode":"Block"},"status"
:{}}]'
    kubevirt.io/latest-observed-api-version: v1
    kubevirt.io/storage-observed-api-version: v1alpha3
    network.harvesterhci.io/ips: '[]'
    sim.harvesterhci.io/creationTimestamp: "2023-02-22T02:04:05Z"
  creationTimestamp: "2023-02-22T02:04:05Z"
  finalizers:
  - wrangler.cattle.io/VMController.UnsetOwnerOfPVCs
  generation: 1
```
Please help see if this still happens after v1.1.2-rc2.

**Comment by bk201** [comment]
@noahgildersleeve If you still have the env, please see if removing the finalizer removes the VM. Thanks!

**Comment by ibrokethecloud** [comment]
should be fixed by: https://github.com/harvester/harvester/pull/3508

**Comment by bk201** [comment]
@noahgildersleeve We can test this with master-head ISO or wait for v1.1.2-rc2

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.1.2-rc3`. Close this issue.

### Result

* The upgrade-repo vm have been deleted after upgrade from v1.1.1 to v1.1.2-rc3
  ![image](https://user-images.githubusercontent.com/29251855/225187953-5e23ce52-b4aa-4773-88a3-99bcdfc92698.png)

* We can delete the existing VM after upgrade on singe node, two nodes and four nodes airgapped upgrade test.
![image](https://user-images.githubusercontent.com/29251855/225188274-04a0843c-d4c9-4d68-a026-29abd61402c0.png)


### Test Information
* Test Environment: 1, 2 and 4 nodes Harvester on kvm and bare machines
* Harvester version: v1.1.2-rc3

### Verify Steps
1. Provision a single or multi nodes Harvester cluster 
2. Provision an RKE2 guest cluster on it
3. Create a new VM
4. offline upgrade to v1.1.2-rc3
5. Start the off VM 
6. Check the VM state after upgrade 
7. Delete the existing VM
